In [41]:
import base64
import requests
from datetime import datetime

base_url = 'https://api.twitter.com/'
client_key = 'a8LPL4fwytcWXdEl0ZvxNu3KN'
client_secret = 'q42fBpSqXuBUwktRvzruW2WILXPIxkJfLVigOVyk4kA0Igw1Gl'

# Getting the OAuth token

In [2]:
key_secret = '{}:{}'.format(client_key, client_secret).encode('ascii')
b64_encoded_key = base64.b64encode(key_secret)
b64_encoded_key = b64_encoded_key.decode('ascii')

In [3]:
auth_url = base_url + 'oauth2/token'

In [4]:
auth_headers = {
    'Authorization': ('Basic '+b64_encoded_key),
    'Content-Type': 'application/x-www-form-urlencoded;charset=UTF-8'
}

auth_data = {
    'grant_type': 'client_credentials'
}

auth_resp = requests.post(auth_url, headers=auth_headers, data=auth_data)

In [5]:
access_token = auth_resp.json()['access_token']

# Searching for tweets

In [129]:
search_headers = {
    'Authorization': ('Bearer '+access_token)    
}

search_params = {
    'q': '#Burger',
#     'result_type': 'recent',
    'count': 500,
    'tweet_mode' : 'extended'
}

search_url = base_url+'1.1/search/tweets.json'

search_resp = requests.get(search_url, headers=search_headers, params=search_params)

In [130]:
search_resp.status_code

200

In [131]:
search_resp.json()

{'statuses': [{'created_at': 'Thu Jan 23 06:22:01 +0000 2020',
   'id': 1220230184122626048,
   'id_str': '1220230184122626048',
   'full_text': '#Celebrating 80 #years of the #Caped #Crusader, The Knight of the Night is at #Burger #King. #lka #iContactLanka https://t.co/EmH4NjwsA1',
   'truncated': False,
   'display_text_range': [0, 112],
   'entities': {'hashtags': [{'text': 'Celebrating', 'indices': [0, 12]},
     {'text': 'years', 'indices': [16, 22]},
     {'text': 'Caped', 'indices': [30, 36]},
     {'text': 'Crusader', 'indices': [37, 46]},
     {'text': 'Burger', 'indices': [78, 85]},
     {'text': 'King', 'indices': [86, 91]},
     {'text': 'lka', 'indices': [93, 97]},
     {'text': 'iContactLanka', 'indices': [98, 112]}],
    'symbols': [],
    'user_mentions': [],
    'urls': [],
    'media': [{'id': 1220230169711071232,
      'id_str': '1220230169711071232',
      'indices': [113, 136],
      'media_url': 'http://pbs.twimg.com/media/EO8g39HX0AATmj7.jpg',
      'media_url_h

In [132]:
tweets_dict = search_resp.json()

# Exploring the results 

In [133]:
tweets_dict.keys()

dict_keys(['statuses', 'search_metadata'])

In [134]:
len(tweets_dict['statuses'])

100

In [135]:
tweets_dict['statuses'][0].keys()

dict_keys(['created_at', 'id', 'id_str', 'full_text', 'truncated', 'display_text_range', 'entities', 'extended_entities', 'metadata', 'source', 'in_reply_to_status_id', 'in_reply_to_status_id_str', 'in_reply_to_user_id', 'in_reply_to_user_id_str', 'in_reply_to_screen_name', 'user', 'geo', 'coordinates', 'place', 'contributors', 'is_quote_status', 'retweet_count', 'favorite_count', 'favorited', 'retweeted', 'possibly_sensitive', 'lang'])

In [136]:
(tweets_dict['statuses'][10])['metadata']

{'iso_language_code': 'es', 'result_type': 'recent'}

## Filterting English tweets

In [137]:
filtered_tweets = []

for tweet in tweets_dict['statuses']:
    language = tweet['metadata']['iso_language_code']
    
    if(language == 'en'):
        filtered_tweets.append(tweet)

In [138]:
len(filtered_tweets)

85

In [139]:
len(tweets_dict['statuses'])

100

In [140]:
filtered_tweets[20]['full_text']

'1/3lb southern fried chicken burger on a ciabatta bun with Jalapeno cheddar, sweet Sriracha, baby spinach and sauteed mushrooms. #burger #nom'

In [141]:
filtered_tweets[20].keys()

dict_keys(['created_at', 'id', 'id_str', 'full_text', 'truncated', 'display_text_range', 'entities', 'metadata', 'source', 'in_reply_to_status_id', 'in_reply_to_status_id_str', 'in_reply_to_user_id', 'in_reply_to_user_id_str', 'in_reply_to_screen_name', 'user', 'geo', 'coordinates', 'place', 'contributors', 'is_quote_status', 'retweet_count', 'favorite_count', 'favorited', 'retweeted', 'lang'])

In [158]:
a = filtered_tweets[15]['retweeted_status']['full_text']

In [159]:
str(a)

'1/2lb all natural beef burger on lettuce blend greens with brie, garlic aioli, red onions and beef chili. #burger #nom'

In [144]:
filtered_tweets[20]['retweeted']

False

In [145]:
filtered_tweets[20]['entities']

{'hashtags': [{'text': 'burger', 'indices': [129, 136]},
  {'text': 'nom', 'indices': [137, 141]}],
 'symbols': [],
 'user_mentions': [],
 'urls': []}

In [146]:
filtered_tweets[3]['full_text']

'RT @mmandmp_pro: A classic burger at my hotel, bought with my VIP £10 voucher, total cost £2.50\n\nPhoto taken with iPhone XS Max \n\n@GoodHote…'

In [147]:
filtered_tweets[3]['full_text']

'RT @mmandmp_pro: A classic burger at my hotel, bought with my VIP £10 voucher, total cost £2.50\n\nPhoto taken with iPhone XS Max \n\n@GoodHote…'

In [148]:
filtered_tweets[3]['retweeted']

False

In [149]:
filtered_tweets[3].keys()

dict_keys(['created_at', 'id', 'id_str', 'full_text', 'truncated', 'display_text_range', 'entities', 'metadata', 'source', 'in_reply_to_status_id', 'in_reply_to_status_id_str', 'in_reply_to_user_id', 'in_reply_to_user_id_str', 'in_reply_to_screen_name', 'user', 'geo', 'coordinates', 'place', 'contributors', 'retweeted_status', 'is_quote_status', 'retweet_count', 'favorite_count', 'favorited', 'retweeted', 'lang'])

In [150]:
filtered_tweets[3]['user']

{'id': 1218522531323416581,
 'id_str': '1218522531323416581',
 'name': 'paul woodman',
 'screen_name': 'mmandmp_12',
 'location': '',
 'description': '',
 'url': None,
 'entities': {'description': {'urls': []}},
 'protected': False,
 'followers_count': 2,
 'friends_count': 17,
 'listed_count': 0,
 'created_at': 'Sat Jan 18 13:16:42 +0000 2020',
 'favourites_count': 182,
 'utc_offset': None,
 'time_zone': None,
 'geo_enabled': False,
 'verified': False,
 'statuses_count': 352,
 'lang': None,
 'contributors_enabled': False,
 'is_translator': False,
 'is_translation_enabled': False,
 'profile_background_color': 'F5F8FA',
 'profile_background_image_url': None,
 'profile_background_image_url_https': None,
 'profile_background_tile': False,
 'profile_image_url': 'http://pbs.twimg.com/profile_images/1218522648604483586/B6r8fUD2_normal.jpg',
 'profile_image_url_https': 'https://pbs.twimg.com/profile_images/1218522648604483586/B6r8fUD2_normal.jpg',
 'profile_link_color': '1DA1F2',
 'profile_sid

In [151]:
tweetedTime = filtered_tweets[3]['created_at']

In [152]:
tweetedTime

'Thu Jan 23 06:09:09 +0000 2020'

In [153]:
tweetedTime[-4:]

'2020'

In [154]:
tweetedTime[4:19]

'Jan 23 06:09:09'

In [155]:
tweetedTime = tweetedTime[4:19] + ' ' + tweetedTime[-4:]
tweetedTime

'Jan 23 06:09:09 2020'

In [156]:
tweetedDateTime = datetime.strptime(tweetedTime, '%b %d %H:%M:%S %Y')
tweetedDateTime

datetime.datetime(2020, 1, 23, 6, 9, 9)

In [161]:
i = 0
for tweet in filtered_tweets:
    print(i)
    try:
        print(str(tweet['retweeted_status']['full_text']))
    except:
        print(str(tweet['full_text']))
        
    print('\n\n')
    i += 1

0
#Celebrating 80 #years of the #Caped #Crusader, The Knight of the Night is at #Burger #King. #lka #iContactLanka https://t.co/EmH4NjwsA1



1
Sometimes you just need a #burger. Ours are made from locally sourced fresh #beef. 01798-812244 RH20 2JN 15 mins #Horsham #Worthing #FollowPierre #greatfood #vegan menu #veganuary #ginbar #restaurant #bestpub #westsussex #dogfriendly #westchiltington #sussexvillage https://t.co/AbRTXcK2GV



2
1lb all natural beef burger on lettuce blend greens with American cheese, horseradish aioli, baby spinach and avocado. #burger #nom



3
A classic burger at my hotel, bought with my VIP £10 voucher, total cost £2.50

Photo taken with iPhone XS Max 

@GoodHotelLondon @hotelsdotcom #goldmember #vip #burger #classicburger 

@fumi_BJ @Yumi_a_BJ @nyu_n62 please follow these ladies, they are super nice.

@BANZAIJAPANPJT https://t.co/iP6tcZKdKo



4
Homemade Cheese Pepperbeef Burgers with barbecue onion sauce. 😋🍔#homemade #homemadefood #homemadecooking #homemade

## Tweet Cleaning

In [215]:
tweetInFocus = filtered_tweets[6]['full_text']
tweetInFocus

'We welcome you on Thursday at #Miam to enjoy our BD 2 offer on all burger and pasta meals! Hurry up and enjoy the offer while it lasts 🍔🍝😍\n\n#miambahrain #restaurant #food #pasta #burger #offer #bahrain #ksa \n #السعودية https://t.co/bauyEqjDMq'

In [168]:
tweetInFocus.split()

['We',
 'welcome',
 'you',
 'on',
 'Thursday',
 'at',
 '#Miam',
 'to',
 'enjoy',
 'our',
 'BD',
 '2',
 'offer',
 'on',
 'all',
 'burger',
 'and',
 'pasta',
 'meals!',
 'Hurry',
 'up',
 'and',
 'enjoy',
 'the',
 'offer',
 'while',
 'it',
 'lasts',
 '🍔🍝😍',
 '#miambahrain',
 '#restaurant',
 '#food',
 '#pasta',
 '#burger',
 '#offer',
 '#bahrain',
 '#ksa',
 '#السعودية',
 'https://t.co/bauyEqjDMq']

In [169]:
len(tweetInFocus.split())

39

## Exploring Emojis

In [173]:
emojiStr = tweetInFocus.split()[28]
emojiStr

'🍔🍝😍'

In [216]:
ord(emojiStr[2])

128525

In [244]:
chr(128506)

'🗺'

In [254]:
for i in range(128512, 128591):
    print(str(i) + ' - ' + chr(i))

128512 - 😀
128513 - 😁
128514 - 😂
128515 - 😃
128516 - 😄
128517 - 😅
128518 - 😆
128519 - 😇
128520 - 😈
128521 - 😉
128522 - 😊
128523 - 😋
128524 - 😌
128525 - 😍
128526 - 😎
128527 - 😏
128528 - 😐
128529 - 😑
128530 - 😒
128531 - 😓
128532 - 😔
128533 - 😕
128534 - 😖
128535 - 😗
128536 - 😘
128537 - 😙
128538 - 😚
128539 - 😛
128540 - 😜
128541 - 😝
128542 - 😞
128543 - 😟
128544 - 😠
128545 - 😡
128546 - 😢
128547 - 😣
128548 - 😤
128549 - 😥
128550 - 😦
128551 - 😧
128552 - 😨
128553 - 😩
128554 - 😪
128555 - 😫
128556 - 😬
128557 - 😭
128558 - 😮
128559 - 😯
128560 - 😰
128561 - 😱
128562 - 😲
128563 - 😳
128564 - 😴
128565 - 😵
128566 - 😶
128567 - 😷
128568 - 😸
128569 - 😹
128570 - 😺
128571 - 😻
128572 - 😼
128573 - 😽
128574 - 😾
128575 - 😿
128576 - 🙀
128577 - 🙁
128578 - 🙂
128579 - 🙃
128580 - 🙄
128581 - 🙅
128582 - 🙆
128583 - 🙇
128584 - 🙈
128585 - 🙉
128586 - 🙊
128587 - 🙋
128588 - 🙌
128589 - 🙍
128590 - 🙎


#### ASCII values from:
1. 128512 - 128526 represent potentially happy emojis 
2. 128527 - 128534 and 128542 - 128561 represent potentially sad emojis

### Text Cleaning

In [256]:
textEntities = tweetInFocus.split()
textEntities

['We',
 'welcome',
 'you',
 'on',
 'Thursday',
 'at',
 '#Miam',
 'to',
 'enjoy',
 'our',
 'BD',
 '2',
 'offer',
 'on',
 'all',
 'burger',
 'and',
 'pasta',
 'meals!',
 'Hurry',
 'up',
 'and',
 'enjoy',
 'the',
 'offer',
 'while',
 'it',
 'lasts',
 '🍔🍝😍',
 '#miambahrain',
 '#restaurant',
 '#food',
 '#pasta',
 '#burger',
 '#offer',
 '#bahrain',
 '#ksa',
 '#السعودية',
 'https://t.co/bauyEqjDMq']

In [ ]:
hashs = []
for i in range(len(textEntities)):
    